api/node/server/server_test.go

```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/spacemeshos/go-spacemesh/activation"
+	types "github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/p2p/pubsub/mocks"
+)
+
+func TestServerReadiness(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	beacons := NewMockbeaconService(ctrl)
+	srv := NewServer(
+		activation.NewMockAtxService(ctrl),
+		beacons,
+		mocks.NewMockPublisher(ctrl),
+		NewMockpoetDB(ctrl),
+		NewMockhare(ctrl),
+		NewMockproposalBuilder(ctrl),
+		zaptest.NewLogger(t),
+	)
+
+	listener, err := net.Listen("tcp", "localhost:0")
+	require.NoError(t, err)
+
+	readiness := make(chan struct{})
+	server := &http.Server{
+		Handler: srv.IntoHandler(http.NewServeMux(), readiness),
+	}
+
+	go server.Serve(listener)
+	defer server.Close()
```

Comment:
Why not use the `Start` method that existed earlier? Also there is no guarantee that `server.Serve` actually terminates. Instead maybe we can do something like:

```
var eg errgroup.Group
eg.Go(func() error {
	return server.Start() // is blocking afaik?
})
t.Cleanup(func() {
	server.Close()
	assert.NoError(t, eg.Wait())
})
```

Reply:
Because it wasn't useful. It would be hard to stop the server with it (`Close()` is a method of the `http.Server`)

Reply:
I don't insist on the `Start` split, but the argument regarding the untracked go routine still holds 🙂